### PR TITLE
[Improve][Connector-V2][Iceberg] Unified exception for iceberg source connector

### LINKED ISSUE #3532: [Improve][Connector-V2][Iceberg] Unified exception for iceberg source connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Improve][Connector-V2][Iceberg] Unified exception for iceberg source connector

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of this pull request

sub-task of #2930 

close #3532 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDeserializer.java [hunk 1]
```diff
@@ -133,7 +135,9 @@ private Object convert(@NonNull Type icebergType,
                 }
                 return seatunnelMap;
             default:
-                throw new UnsupportedOperationException("Unsupported iceberg type: " + icebergType);
+                throw new IcebergConnectorException(
+                    CommonErrorCode.UNSUPPORTED_OPERATION,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDeserializer.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE

**Comment by hailin0** [comment]
waiting for this pr fix e2e-testcase https://github.com/apache/incubator-seatunnel/pull/2865

**Review by TyrantLucifer — DISMISSED** [review]

**Comment by TyrantLucifer** [comment]
> waiting for this pr fix e2e-testcase #2865

What the status of this task? @hailin0

**Review by hailin0 — DISMISSED** [review]

**Comment by hailin0** [comment]
> > waiting for this pr fix e2e-testcase #2865
> 
> What the status of this task? @hailin0

@TyrantLucifer Please review

**Comment by hailin0** [comment]
> waiting for this pr fix e2e-testcase #2865

image

**Review by TyrantLucifer — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**